pkg/controller/shoot/shoot_maintenance_control.go

```diff
@@ -75,23 +78,75 @@ func (c *Controller) reconcileShootMaintenanceKey(key string) error {
 		return nil
 	}
 	if err != nil {
-		logger.Logger.Infof("[SHOOT MAINTENANCE] %s - unable to retrieve object from store: %v", key, err)
+		logger.Logger.Errorf("[SHOOT MAINTENANCE] %s - unable to retrieve object from store: %v", key, err)
```

Comment:
Should we add some dedicated handling for the case that `apierrors.IsNotFound(err)`?

Reply:
There is already handling for this error. Check line 76.